src/tools/compiletest/src/lib.rs

```diff
@@ -1029,11 +1023,11 @@ fn make_test_closure(
 /// To avoid problems, we forbid test names from overlapping in this way.
 ///
 /// See <https://github.com/rust-lang/rust/pull/109509> for more context.
-fn check_overlapping_tests(found_paths: &HashSet<PathBuf>) {
+fn check_for_overlapping_test_paths(found_path_stems: &BTreeSet<PathBuf>) {
```

Comment:
There is actually perf difference, https://github.com/rust-lang/rust/pull/120273/commits/0710ebb999dd633bd536e66fcaab3d2a4b1739e8

Reply:
Ahh, that was linked before https://github.com/rust-lang/rust/pull/131870#discussion_r1805939873

Reply:
The PR description said "Wall time didn't improved :-) ."

Reply:
Lol, I'm going to rerun some benchmarks locally